CONTRIBUTING.md

```diff
@@ -13,14 +13,62 @@ Please fill out either the individual or corporate Contributor License Agreement
 
 Follow either of the two links above to access the appropriate CLA and instructions for how to sign and return it. Once we receive it, we'll be able to accept your pull requests.
 
+## Composition of This Repository and Where/How to Contribute
+
+The repository of the Kubernetes Python client consists of this main repository and a submodule, the [python-base](https://github.com/kubernetes-client/python-base) repository. The main repository contains mostly files that are generated by the OpenAPI generator from [this OpenAPI spec](scripts/swagger.json). The base repo is the utility part of the python client and allows developers to create their own kubernetes clients.
+
+### Where to Submit Your Patch
+
+These folders are automatically generated. You will need to submit a patch to the upstream kubernetes repo [kubernetes](https://github.com/kubernetes/kubernetes) or the OpenAPI generator repo [openapi-generator](https://github.com/OpenAPITools/openapi-generator). This contains:
+[kubernetes/client](kubernetes/client), [kubernetes/test](kubernetes/test), [kubernetes/docs](kubernetes/docs).
+
+These folders contain developer written codes in the main repo and the patches should be submitted here:
+[kubernetes/e2e_test](kubernetes/e2e_test), [kubernetes/utils](kubernetes/utils), [examples](examples), [scripts](scripts).
+
+The base repo contains developer written codes only. The patches should be sent to the base repo instead. The scope covers these repos and symbolic links in the main repo:
+kubernetes/base, kubernetes/config, kubernetes/stream, kubernetes/watch.
```

Comment:
could you add a paragraph about "how to update the base submodule in the main repo after your python-base PR is merged":
```
# In your local branch of the python repo
$ cd kubernetes/base
$ git fetch origin
$ git checkout origin/master # assuming your change has been merged into the latest python-base MASTER
$ cd ../..
# Add release notes for you base-repo change in CHANGELOG.md
# Finally, commit and create a PR against the python repo
```